test/sharness/lib/test-lib.sh

```diff
@@ -353,3 +355,38 @@ test_check_peerid() {
 	}
 }
 
+# A workaround for https://github.com/ipfs/go-ipfs/issues/1941 ‘sharness suite:
+# ipfs name publish: “Error: failed to find any peer in table”’.  Certain
+# commands fail whenever the daemon is running with no peers.
+# test_expect_success_1941 runs test_expect_failure if that is the case and
+# test_expect_success otherwise.  As soon as #1941 is fixed, all invocations of
+# test_expect_success_1941 can be replaced with test_expect_success and this
+# function can be removed.
```

Comment:
I’m not sure what to do other than having #1941 block this PR. These tests will simply fail until it is fixed. Previously there were duplicated blocks of tests where the succeeding (offline) tests used `test_expect_success` and the failing (online) tests used `test_expect_failure` but that was not acceptable. Now both use the same `test_expect_success_1941` which is *temporary* and will be globally replaced with `test_expect_success` as soon as #1941 is fixed.

#1941 blocking this PR would be less than optimal because the code *works*, the testing environment just fails to do IPNS right. (And if you run the tests without the `ipfs bootstrap rm --all`, they all pass.)